maraboupy/MarabouNetworkNNet.py

```diff
@@ -263,31 +563,346 @@ def buildEquations(self):
                     # Adjust weights of output layer to incorporate output normalization
                     elif self.normalize and layer == len(self.layerSizes) - 1:
                         weight *= self.outputRange
-                    e.addAddend(weight, self.nodeTo_f(layer-1, previous_node))
+                    e.addAddend(weight, self.nodeTo_f(layer - 1, previous_node))
 
                 # Adjust bias of output layer to incorporate output normalization
                 if self.normalize and layer == len(self.layerSizes) - 1:
                     bias = bias * self.outputRange + self.outputMean
                 e.setScalar(-bias)
                 self.addEquation(e)
 
-    """
-    Identify all relus and their associated variable numbers and add them to Marabou network
-    Arguments:
-        None
-    """
     def addRelus(self):
+        """
+        Identify all relus and their associated variable numbers and add them to Marabou network
+        Arguments:
+            None
+        """
         relus = []
         hidden_layers = self.layerSizes[1:-1]
         for layer, size in enumerate(hidden_layers):
             for node in range(size):
-                self.addRelu(self.nodeTo_b(layer+1, node), self.nodeTo_f(layer+1, node))
+                self.addRelu(self.nodeTo_b(layer + 1, node), self.nodeTo_f(layer + 1, node))
 
     def numberOfVariables(self):
-        return self.layerSizes[0] + 2*sum(self.layerSizes[1:-1]) + 1*self.layerSizes[-1]
+        return self.layerSizes[0] + 2 * sum(self.layerSizes[1:-1]) + 1 * self.layerSizes[-1]
 
     def getInputMinimum(self, inputVar):
         return self.inputMinimums[inputVar]
 
     def getInputMaximum(self, inputVar):
         return self.inputMaximums[inputVar]
+
+
+    def evaluateNetwork(self, inputs, normalize_inputs=True, normalize_outputs=True, activate_output_layer=False):
```

Comment:
As mentioned in a previous comment, this function should be named "evaluateWithoutMarabou". Variables normalize_inputs and normalize_outputs should be changed to self.normalize, and activate_output_layer is not necessary as the NNet format assumes activation only on the hidden layers, not output layer. This function should only take "inputs" as the argument (besides "self").

Reply:
Inputs to this function such as first_layer and last_layer that default to -1, for example, would allow you to combine this function and the to/from layer evaluation functions into one.

Reply:
I have now combined all the evaluations in one method evaluateNNet, and moved evaluateWithoutMarabou (which calls evaluateNNet) from MarabouNetwork to MarabouNetworkNNet, as you have suggested. 

The reason I want to have the additional features has to do with a project that I am working on (invariant search), which involves looking at the behavior of different hidden layers, and evaluating the network to a from a certain layer, as well as splitting the network into two (I have a function that does this for an nnet, which I did not include in the PR). In particular, sometimes I want to activate the output layer (of a "sliced" network) in order to get correct results.

Reply:
I see, thanks for the explanation! A function to split the network into two might be a useful feature for a subsequent pr.

Reply:
Guy thought that it did not belong in the master at this point, this is why I excluded it from this PR. :) But if you think that it might be useful, perhaps I will still include it in the next one.